distributed/scheduler.py

```diff
@@ -4459,6 +4472,12 @@ def update_graph(
                     recommendations[ts.key] = "erred"
                     break
 
+        annotations_for_plugin: defaultdict[str, dict[str, Any]] = defaultdict(dict)
+        for key in keys_with_annotations:
+            ts = self.tasks[key]
+            for annot, value in ts.annotations.items():
+                annotations_for_plugin[annot][key] = value
```

Comment:
I'm inclined to remove passing annotations to plugins. If a scheduler plugin wants to know what annotations are there, it should just look this up with the task itself. I wanted to separate this change from the refactoring so I decided to keep a slightly simplified version in